File: HeartForCharity/HeartForCharity.Services/Database/Review.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace HeartForCharity.Services.Database
{
    public class Review
    {
        [Key]
        public int ReviewId { get; set; }

        [Required]
        [ForeignKey(nameof(VolunteerApplication))]
        public int VolunteerApplicationId { get; set; }

        [Required]
        [ForeignKey(nameof(OrganisationProfile))]
        public int OrganisationProfileId { get; set; }

        [Required]
        [ForeignKey(nameof(UserProfile))]
        public int UserProfileId { get; set; }

        [Required]
        [Range(1, 5)]
        public int Rating { get; set; }

        [MaxLength(2000)]
        public string? Comment { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? DeletedAt { get; set; }

        // Navigacijska svojstva
        public virtual VolunteerApplication VolunteerApplication { get; set; } = null!;
        public virtual OrganisationProfile OrganisationProfile { get; set; } = null!;
        public virtual UserProfile UserProfile { get; set; } = null!;
    }
}

File: HeartForCharity/UI/heartforcharity_mobile/lib/model/responses/review.dart
class Review {
  final int reviewId;
  final int? volunteerApplicationId;
  final int? organisationProfileId;
  final String? organisationName;
  final int? userProfileId;
  final String? reviewerName;
  final int rating;
  final String? comment;
  final DateTime? createdAt;

  Review({
    this.reviewId = 0,
    this.volunteerApplicationId,
    this.organisationProfileId,
    this.organisationName,
    this.userProfileId,
    this.reviewerName,
    this.rating = 0,
    this.comment,
    this.createdAt,
  });

  factory Review.fromJson(Map<String, dynamic> json) => Review(
        reviewId: json['reviewId'] ?? 0,
        volunteerApplicationId: json['volunteerApplicationId'],
        organisationProfileId: json['organisationProfileId'],
        organisationName: json['organisationName'],
        userProfileId: json['userProfileId'],
        reviewerName: json['reviewerName'],
        rating: json['rating'] ?? 0,
        comment: json['comment'],
        createdAt: json['createdAt'] != null ? DateTime.parse(json['createdAt']) : null,
      );
}
